clevertap-core/src/test/java/com/clevertap/android/sdk/network/api/CtApiTest.kt

```diff
@@ -242,15 +242,29 @@ class CtApiTest {
             customHandshakeDomain = null
         }
 
+        var counter = 0
```

Comment:
You can replace the counter logic with mockk' verify methods:

```kotlin
        val clearCachedHandshakeDomain = mockk<() -> Unit>(relaxed = true)

        // assert
        assertEquals(
            expected = false,
            actual = ctApi.needsHandshake(false, clearCachedHandshakeDomain)
        )
        verify(exactly = 1) { clearCachedHandshakeDomain() }

        assertEquals(
            expected = false,
            actual = ctApi.needsHandshake(true, clearCachedHandshakeDomain)
        )
        verify(exactly = 2) { clearCachedHandshakeDomain() }
```

Reply:
the logic is removed in the chained pr as we no longer need a lambda, thanks for the hints, will remember in next set of UTs.